src/pages/PathUpdate.vue

```diff
@@ -0,0 +1,58 @@
+<script lang="ts">
+import Vue from 'vue';
+import { ipcRenderer } from 'electron';
+import { mapGetters } from 'vuex';
+import PathManagementSelector from '~/components/PathManagementSelector.vue';
+import { PathManagementStrategy } from '~/integrations/pathManager';
+
+export default Vue.extend({
+  name:       'path-update',
+  components: { PathManagementSelector },
+  layout:     'dialog',
+  computed:   { ...mapGetters('applicationSettings', { pathManagementStrategy: 'getPathManagementStrategy' }) },
+  beforeMount() {
+    window.addEventListener('beforeunload', this.commitStrategy);
+  },
+  mounted() {
+    ipcRenderer.send('dialog/ready');
+  },
+  beforeDestroy() {
+    window.removeEventListener('beforeunload', this.commitStrategy);
+  },
+  methods: {
+    setPathManagementStrategy(val: PathManagementStrategy) {
+      this.$store.dispatch('applicationSettings/setPathManagementStrategy', val);
+    },
+    async commitStrategy() {
+      await this.$store.dispatch('applicationSettings/commitPathManagementStrategy', this.pathManagementStrategy);
+      ipcRenderer.send('window/close');
```

Comment:
Given that we have _already_ committed the strategy, why do we need to roundtrip back to the main process to close this window? Why not just call `window.close()` directly?

Reply:
Yeah, I think that makes sense. I was originally anticipating having more to do in the main process, but I think I ended up offloading a lot of that work to the renderer anyway.